lib/conjur/cidr.rb

```diff
@@ -0,0 +1,29 @@
+require 'ipaddr'
+
+module Conjur
+  class CIDR
+    def initialize(cidr)
+      @cidr = cidr.to_s
+      @ipaddr = cidr.kind_of?(IPAddr) ? cidr : IPAddr.new(cidr)
```

Comment:
I think rather than having flexible input here, I'd rather make the client be specific:

1. canonical `new` ctor for IPAddrs
2. a `from_s` ctor for strings, which converts to IP and then delegates to 1.

Reply:
@jonahx, that actually doesn't work well in this case because string to `IPAddr` is lossy (where it strips the extra bits in an invalid CIDR input). We need to preserve the original string representation if we have one.

In this particular case, we don't actually need to accept input as an `IPAddr`, so I'll just remove that as YAGNI. If we find we need it later, we can the additional constructor then.